### PR TITLE
PTX-6403: fix nil panic caused by nil parameter

### PR DESCRIPTION
**Pull request by trierra**
the nil parameter in line causes nil panic later in the thread, fixed the parameter check 

 
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by nikolaypopov — APPROVED** [review]
LGTM

**Review by dahuang-purestorage — APPROVED** [review]
LGTM

File: drivers/scheduler/k8s/k8s.go [hunk 1]
```diff
@@ -2613,10 +2613,14 @@ func (k *K8s) ValidateVolumes(ctx *scheduler.Context, timeout, retryInterval tim
...
 		} else if obj, ok := specObj.(*v1.PersistentVolumeClaim); ok {
 			err := k8sCore.ValidatePersistentVolumeClaim(obj, timeout, retryInterval)
-			if err != nil && !options.ExpectError {
-				return &scheduler.ErrFailedToValidateStorage{
-					App:   ctx.App,
-					Cause: fmt.Sprintf("Failed to validate PVC: %v. Err: %v", obj.Name, err),
+			if err != nil {
+				if options != nil && options.ExpectError {
+					// ignore
+				} else {
```
Comment (piyush-nimbalkar) [inline drivers/scheduler/k8s/k8s.go:2619]: nit: We could also write it like this - 
```
if err != nil && (options == nil || !options.ExpectError) {
    return &scheduler.ErrFailedToValidateStorage{...}
}
```

Comment (trierra) [reply drivers/scheduler/k8s/k8s.go:2619]: yep, I could do that too, it just a bit easier to read for me in this way

**Pull request merged**